### DISCUSSION TITLE
scheda energia e sonoff powr316: avviso "Le seguenti entità non hanno la classe di dispositivo prevista"

### DISCUSSION THREAD

**Discussion by mosix76** [indomus/forum#3049] [HUB: Home Assistant]
Ciao ragazzi,
ho aggiunto alla scheda energia 3 sonoff pow316, con firmware originale. Posso scegliere se selezionare il consumo giornaliero, o mensile. Qualunque sia la scelta, una volta selezionato, nella sezione "Rete elettrica" compare un messaggio di avviso con scritto: "Le seguenti entità non hanno la classe di dispositivo prevista" .
Allego immagine:
![image](https://github.com/indomus/forum/assets/11273690/c946e2ba-8a62-4a50-b93b-f39d42a9b9bd)

Dopo alcune prove, mi sono accorto che i dispositivi vengono cmq aggiunti alla scheda energia, e il consumo mensile credo dia il resoconto corretto. 

Con Tasmota non ho mai avuto questi problemi. Sto sbagliando qualcosa? C'e' modo di impostare la classe energetica per i sonoff pow originali? 

Grazie a tutti

**Comment by Scialla** [comment]
i sensori devono avere queste impostazioni:
```
unit_of_measurement: kWh
state_class: total_increasing
device_class: energy
```
se non ce l'hanno puoi aggiungergliela usando `customize.yaml`
inoltre devono essere presenti nel recorder (e quindi devi aggiungerle e riavviare)
